### ISSUE TITLE
Add public header support for .inc files

### ISSUE THREAD

**Issue by eltonkent** [gradle/gradle-native#1071]
Public header files do not support .inc files. Therefore, the gradle native plugins
 '`cpp-library`', '`c-library`' does not export` .inc` files when publishing libraries.

### Steps to Reproduce (for bugs)
Create a c++ library project with a .inc header file in the public header directory. 
- Publish the library locally.
- The exported public header files do not contain the .inc files.

### Your Environment
Gradle - 6.3
OS - Mac OS X 10.14.6
Compiler - Clang

**Comment by lacasseio** [comment]
You could manually include the files in the publishing zip for the headers to work around your issue. However, the zip task for the header is somewhat internal so it's not ideal.

**Comment by eltonkent** [comment]
I ended up writing a task that overwrites the cpp-headers.zip file. Would be nice if the header extension can be configurable.

**Comment by lacasseio** [comment]
I agree, I will consider it as part of the [Nokee plugins](https://nokee.dev).
